cli/stats.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	containerHeader     = "CONTAINER ID"
+	containerNameHeader = "NAME"
+	cpuPercHeader       = "CPU %"
+	netIOHeader         = "NET I/O"
+	blockIOHeader       = "BLOCK I/O"
+	memPercHeader       = "MEM %"
+	memUseHeader        = "MEM USAGE / LIMIT"
+	pidsHeader          = "PIDS"
+)
+
+// statsDescription is used to describe stats command in detail and auto generate command doc.
+var statsDescription = "stats command is to display a live stream of container(s) resource usage statistics"
+
+// StatsCommand use to implement 'stats' command
+type StatsCommand struct {
+	baseCommand
+
+	noStream bool
+	//TODO: add more flags support
+}
+
+// Init initialize stats command.
+func (stats *StatsCommand) Init(c *Cli) {
+	stats.cli = c
+	stats.cmd = &cobra.Command{
+		Use:   "stats [OPTIONS] [CONTAINER...]",
+		Short: "Display a live stream of container(s) resource usage statistics",
+		Long:  statsDescription,
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return stats.runStats(args)
+		},
+		Example: statsExample(),
+	}
+	stats.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (stats *StatsCommand) addFlags() {
+	flagSet := stats.cmd.Flags()
+	flagSet.BoolVar(&stats.noStream, "no-stream", false, "Disable streaming stats and only pull the first result")
+}
+
+// runStats is the entry of stats command.
+func (stats *StatsCommand) runStats(args []string) error {
+	ctx := context.Background()
+	apiClient := stats.cli.Client()
+	containers := args
+
+	cStats := []*StatsEntryWithLock{}
+	waitFirst := &sync.WaitGroup{}
+	for _, name := range containers {
+		s := &StatsEntryWithLock{StatsEntry: StatsEntry{container: name}}
+		cStats = append(cStats, s)
+		waitFirst.Add(1)
+		go collect(ctx, s, apiClient, !stats.noStream, waitFirst)
+	}
+
+	// before print to screen, make sure each container get at least one valid stat data
+	waitFirst.Wait()
+
+	// do a quick scan in case container not found and so on
```

Comment:
Is this check duplicated with the line 105 one. You can put the `cleanScreen` after the line 105 loop

Reply:
this quick check do scan error before stream the stats api, line 105 check in case the there's runtime error when connect api, the quick check scan make true all containers can use stats api(not forbid by error such as container not find or container state not right).